services/QuillLMS/app/helpers/vitally_teacher_stats_helper.rb

```diff
@@ -0,0 +1,62 @@
+module VitallyTeacherStatsHelper
+  DIAGNOSTIC_ID = 4
+
+  def total_students_in_year(user, school_year_start, school_year_end)
+    all_classrooms = user.classrooms_i_teach + user.archived_classrooms
+    year_classrooms = all_classrooms.select { |c| school_year_start <= c.created_at && school_year_end > c.created_at }
+    year_classrooms.sum { |c| c.students.count}
+  end
+
+  def active_students_query(user)
+    @active_students ||= ActivitySession.select(:user_id).distinct
+    .joins("JOIN classroom_units on classroom_units.id = activity_sessions.classroom_unit_id")
+    .joins("JOIN classrooms_teachers on classrooms_teachers.classroom_id=classroom_units.classroom_id")
+    .where(state: 'finished')
+    .where('classrooms_teachers.user_id = ?', user.id)
+  end
+
+  def activities_assigned_in_year_count(user, school_year_start, school_year_end)
+    sum_students(in_school_year(activities_assigned_query(user), school_year_start, school_year_end))
+  end
+
+  def sum_students(records)
+    records.map { |r| r&.assigned_student_ids&.count || 0 }.sum || 0
```

Comment:
I don't believe it's possible to reach the final `|| 0`, since the sum of an empty enumerable is 0